virtcontainers/qemu.go

```diff
@@ -1179,6 +1180,28 @@ func (q *qemu) saveSandbox() error {
 		return err
 	}
 
+	t := time.NewTimer(qmpMigrationWaitTimeout)
+	defer t.Stop()
+	for {
+		status, err := q.qmpMonitorCh.qmp.ExecuteQueryMigration(q.qmpMonitorCh.ctx)
+		if err != nil {
+			q.Logger().WithError(err).Error("failed to query migration status")
+			return err
+		}
+		if status.Status == "completed" {
+			break
+		}
+
+		select {
+		case <-t.C:
+			q.Logger().WithField("migration-status", status).Error("timeout waiting for qemu migration")
```

Comment:
How is this loop exited on failure? Should this line be something like:

```go
return fmt.Errorf("timed out after %d seconds waiting for qemu migration", qmpMigrationWaitTimeout)
```

Reply:
oops, I missed a return here!

Reply:
@bergwolf Still dont see the return here.

Reply:
oops, pushed wrong branch. Fixed now.